contracts/contracts/strategies/AaveStrategy.sol

```diff
@@ -170,19 +213,71 @@ contract AaveStrategy is InitializableAbstractStrategy {
     }
 
     /**
-     * @dev Get the current address of the Aave lending pool core, which stores all the
-     *      reserve tokens in its vault.
-     * @return Current lending pool core address
+     * @dev Collect stkAave, convert it to AAVE send to Vault.
      */
-    function _getLendingPoolCore() internal view returns (address payable) {
-        address payable lendingPoolCore = ILendingPoolAddressesProvider(
-            platformAddress
-        )
-            .getLendingPoolCore();
-        require(
-            lendingPoolCore != address(uint160(address(0))),
-            "Lending pool core does not exist"
-        );
-        return lendingPoolCore;
+    function collectRewardToken() external onlyVault nonReentrant {
+        if (address(stkAave) == address(0)) {
+            return;
+        }
+
+        // Check staked AAVE cooldown timer
+        uint256 cooldown = stkAave.stakersCooldowns(address(this));
+        uint256 windowStart = cooldown + stkAave.COOLDOWN_SECONDS();
+        uint256 windowEnd = windowStart + stkAave.UNSTAKE_WINDOW();
+
+        // If inside the unlock window, then we can redeem stkAave
+        // for AAVE and send it to the vault.
+        if (now > windowStart && now <= windowEnd) {
+            // Redeem to AAVE
+            uint256 stkAaveBalance = stkAave.balanceOf(address(this));
+            if (stkAaveBalance > rewardLiquidationThreshold) {
+                stkAave.redeem(address(this), stkAaveBalance);
+            }
+            // Transfer AAVE to vaultAddress
+            uint256 aaveBalance = IERC20(rewardTokenAddress).balanceOf(
+                address(this)
+            );
+            if (aaveBalance > 0) {
+                IERC20(rewardTokenAddress).safeTransfer(
+                    vaultAddress,
+                    aaveBalance
+                );
+            }
+        }
+
+        // If we were past the start of the window,
```

Comment:
This comment is a bit confusing. I think it needs to be updated to mention claiming of rewards. "If we have passed the start of the window, claim any outstanding rewards. If the cooldown counter is not running then start it."

Reply:
Updated the comment.